sky/skylet/log_lib.pyi

```diff
@@ -0,0 +1,108 @@
+from sky import sky_logging as sky_logging
```

Comment:
I'm a little new to mypy, but is this stub file necessary if we fully type `log_lib.py`?

Reply:
This stub file is mainly for the type hint of `run_with_log` where we want to determine the return type by the value of the argument `require_outputs`. To remove this stub file, we can move the two `typing.overload` back into the original `log_lib.py`, but that may make that file much longer. Wdyt?

Reply:
Ahh... I think it would be nicer to move the overload definitions to `log_lib.py` (fewer files to deal with, easier to maintain). 

On a related note - should we reconsider why we have different return types for `run_with_log`? It would be nice if we can get away with `Tuple[int, str, str]` and `return proc.returncode, '', ''` when require_outputs is false?

Reply:
(Casual skim, didn't read the discussion in full) `.pyi` files being standalone seems common practice:
- https://github.com/tensorflow/tensorflow/blob/master/tensorflow/compiler/xla/python/xla_client.pyi
- https://github.com/tensorflow/tensorflow/blob/master/tensorflow/compiler/xla/python/xla_extension/ops.pyi
- https://github.com/ray-project/ray/blob/master/python/ray/_raylet.pyi

Reply:
Oops, sorry. Just realized, another reason we have to use the stub file is that the pylint has a bug with the type overriding in the same file ([reference](https://github.com/PyCQA/pylint/issues/5264)). It will not correctly recognize the overrides types and complain about `sky/backends/cloud_vm_ray_backend.py:1362:8: E0633: Attempting to unpack a non-sequence (unpacking-non-sequence)`.
I tried to upgrade the pylint to the latest version but the problem still exists. I had to keep the stub file for now.

I am planning to refactor the `require_outputs` and remove the stub file in another PR as otherwise, the changes will be mixed with the mypy changes in this PR, making it hard to distinguish the changes. : )